### PR TITLE
install: verify no hooks colliding with dvc exist

### LINKED ISSUE #2362: hooks: dvc appends hooks and breaks the existing hook
If git hook is a python file users end up with a broken hook - a mix of python and bash code. We need to find a way to mitigate this.

### PR DESCRIPTION
**Pull request by pared**
* [x] Have you followed the guidelines in our
      [Contributing document](https://dvc.org/doc/user-guide/contributing)?

* [x] Does your PR affect documented changes or does it add new functionality
      that should be documented? If yes, have you created a PR for
      [dvc.org](https://github.com/iterative/dvc.org) documenting it or at
      least opened an issue for it? If so, please add a link to it.

-----
Fixes #2362

### CODE REVIEW

File: tests/func/test_install.py [hunk 1]
```diff
@@ -30,6 +30,13 @@ def test_should_create_hooks(self, git, dvc_repo):
             with open(self._hook(fname), "r") as fobj:
                 assert command in fobj.read()
 
+    def test_should_fail_if_file_already_exists(self, git, dvc_repo):
+        with open(self._hook("post-checkout"), "w") as fobj:
+            fobj.write("hook content")
+
+        assert main(["install"]) != 0
+
+    @pytest.mark.skip(reason="https://github.com/iterative/dvc/issues/2362")
```
Comment (efiop) [inline tests/func/test_install.py]: Leftover from WIP? These tests should be removed now, right?

Comment (pared) [reply tests/func/test_install.py]: Well, to be honest looking at #2362 I though that this change is first step to handle hooks more gracefully. I though that we might want to get back to this tests at some point, and thats why I thought leaving them around might be a good think. What do you think?

Comment (efiop) [reply tests/func/test_install.py]: @pared No need to keep unused code, we can bring this back later if we need to without any problems.

File: tests/func/test_install.py [hunk 2]
```diff
@@ -30,36 +30,11 @@ def test_should_create_hooks(self, git, dvc_repo):
...
-        expected_script = (
-            "#!/bin/sh\n"
-            '[ "$3" = "0" ]'
-            ' || [ -z "$(git ls-files .dvc)" ]'
-            " || exec dvc checkout\n"
-        )
-
-        with open(self._hook("post-checkout"), "r") as fobj:
-            assert fobj.read() == expected_script
+        assert main(["install"]) != 0
```
Comment (efiop) [inline tests/func/test_install.py]: While we are at it, let's keep it more granular.
```suggestion
        with pytest.raises(GitHookAlreadyExistsError):
            dvc_repo.scm.install()
```

**Pull request merged**